src/GraphQL/Instrumentation/TelemetryProvider.cs

```diff
@@ -0,0 +1,135 @@
+
+        // record the operation type and the operation name (which may be specified within the
+        // document even if not specified in the request)
+        options.Listeners.Add(new TelemetryListener(this, activity, options));
+
+        // execute the request
+        var result = await next(options).ConfigureAwait(false);
+
+        // record the status
+        await SetResultTags(activity, options, result).ConfigureAwait(false);
+
+        // return the result
+        return result;
+    }
+
+    /// <summary>
+    /// Creates an <see cref="Activity"/> for the specified <see cref="ExecutionOptions"/> and starts it.
+    /// </summary>
+    protected virtual ValueTask<Activity?> StartActivity(ExecutionOptions options)
+        => new(_activitySource.StartActivity(ACTIVITY_OPERATION_NAME));
+
+    /// <summary>
+    /// Sets the <see cref="Activity"/> tags based on the specified <see cref="ExecutionOptions"/>.
+    /// The default implementation sets the <c>graphql.operation.name</c> and <c>graphql.document</c> tags.
+    /// </summary>
+    protected virtual Task SetInitialTags(Activity activity, ExecutionOptions options)
+    {
+        activity.SetTag("graphql.operation.name", options.OperationName);
+        if (_telemetryOptions.RecordDocument && activity.IsAllDataRequested)
+            activity.SetTag("graphql.document", options.Query);
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Sets the <see cref="Activity"/> tags based on the specified <see cref="GraphQLOperationDefinition"/> and other values.
+    /// The default implementation sets the <c>graphql.operation.name</c> and <c>graphql.operation.type</c> tags, and sets the
+    /// <see cref="Activity.DisplayName"/> property based on the operation name and type.
+    /// </summary>
+    protected virtual Task SetOperationTags(Activity activity, ExecutionOptions options, ISchema schema, GraphQLDocument document, GraphQLOperationDefinition operation)
+    {
+        var operationType = operation.Operation.ToString().ToLowerInvariant();
+        activity.SetTag("graphql.operation.type", operationType);
+        var operationName = operation.Name?.StringValue;
+        activity.SetTag("graphql.operation.name", operationName);
+        activity.DisplayName = operationName == null ? operationType : $"{operationType} {operationName}";
+        return Task.CompletedTask;
+    }
+
+    /// <summary>
+    /// Sets the <see cref="Activity"/> tags based on the specified <see cref="ExecutionResult"/>.
+    /// The default implementation for .NET 6+ sets the status code tag based on the <see cref="ExecutionResult.Errors"/> property.
+    /// </summary>
+    protected virtual Task SetResultTags(Activity activity, ExecutionOptions executionOptions, ExecutionResult result)
+    {
+#if NET6_0_OR_GREATER
+        var failed = result.Errors?.Count > 0;
+        activity.SetStatus(failed ? ActivityStatusCode.Error : ActivityStatusCode.Ok);
```

Comment:
> It says
> 
> > When an exception is handled, a Span status can be set to Error. Otherwise, a Span status is in the Unset state.

I have checked all implementations in OpenTelemetry and OpenTelemetry.Contrib repository. If I do not missed anything all of them are setting only Errors. Ok status is not explicitly set.

Reply:
Reference: https://github.com/open-telemetry/opentelemetry-specification/blob/v1.16.0/specification/trace/api.md?plain=1#L598-L600

Reply:
Makes sense.  Again, if we do set status to Ok, then we can determine when the client disconnects since there will be no Ok or Error.

I would like @sungam3r 's opinion on this also.

Reply:
Current code is against the OTel Specification recommendation.

> Instrumentation Libraries SHOULD NOT set the status code to Ok,
unless **explicitly** configured to do so.

Reply:
> Application developers and Operators

Application developer is the user of `GraphQL` library. Not the [instrumentation] library author. These are different roles.

Reply:
Thanks for the clarification.  When we have time to come back to this, we will review those specs in detail.

Reply:
> I have checked all implementations in OpenTelemetry and OpenTelemetry.Contrib repository. If I do not missed anything all of them are setting only Errors. Ok status is not explicitly set.

Yep, it looks like we should follow the same way here.

Reply:
ok....